lib/spack/spack/package.py

```diff
@@ -1245,43 +1245,68 @@ def do_fake_install(self):
         dump_packages(self.spec, packages_dir)
 
     def _if_make_target_execute(self, target):
-        try:
-            # Check if we have a makefile
-            file = [x for x in ('Makefile', 'makefile') if os.path.exists(x)]
-            file = file.pop()
-        except IndexError:
+        make = inspect.getmodule(self).make
+
+        # Check if we have a Makefile
+        for makefile in ['GNUmakefile', 'Makefile', 'makefile']:
+            if os.path.exists(makefile):
+                break
+        else:
             tty.msg('No Makefile found in the build directory')
             return
```

Comment:
This block isn't actually needed anymore, it is only used in the final output message. Do you think it's still worth keeping?

Reply:
What about something like:
```python
if not any(os.path.exists(makefile) for makefile in ['GNUmakefile', 'Makefile', 'makefile']):
    tty.msg('No Makefile found in the build directory')
    return
```
?

Reply:
I personally find this more readable, and you suggestion will need to be split onto multiple lines anyway. Plus, this results in `makefile` being set to the Makefile name, whereas I don't know if your suggestion does.

Reply:
Fair enough. Readability is subjective.